packages/core/src/common/abstractComponent.ts

```diff
@@ -21,6 +26,31 @@ export abstract class AbstractComponent<P, S> extends React.Component<P, S> {
         this.validateProps(nextProps);
     }
 
+    public componentWillUnmount() {
+        this.clearTimeouts();
+    }
+
+    /**
+     * Set a timeout and remember its ID.
+     * All stored timeouts will be cleared when component unmounts.
+     * @returns a "cancel" function that will clear timeout when invoked.
+     */
+    public setTimeout(handler: Function, timeout?: number) {
+        const handle = setTimeout(handler, timeout);
+        this.timeoutHandles.push(handle);
+        return () => clearTimeout(handle);
+    }
+
+    /**
+     * Clear all known timeouts.
+     */
+    public clearTimeouts = () => {
+        if (this.timeoutHandles.length > 0) {
+            // always produces empty list cuz clearTimeout returns void
```

Comment:
then... why is it a filter? usage of filter just makes the code convoluted (the fact that you added this comment also suggests that). I would prefer to do:

```ts
for (const timeoutId of this.timeoutIds) {
    clearTimeout(timeoutId);
}
this.timeoutIds = [];
```